src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -504,6 +500,9 @@ CHIP_ERROR UDPEndPointImplLwIP::IPv6JoinLeaveMulticastGroupImpl(InterfaceId aInt
 #ifdef HAVE_IPV6_MULTICAST
     const ip6_addr_t lIPv6Address = aAddress.ToIPv6();
     err_t lStatus;
+
+    ScopedLwIPLock lwipLock;
```

Comment:
Again, would be better to scope as narrowly as possible.

Reply:
Narrowed the scope.